eng/scripts/tests/Invoke-GenerateAndBuildV2.tests.ps1

```diff
@@ -0,0 +1,52 @@
+#Requires -Version 7.0
+. (Join-Path $PSScriptRoot ".." ".." "common" "scripts" "Helpers" PSModule-Helpers.ps1)
+
+Install-ModuleIfNotInstalled "Pester" "5.3.3" | Import-Module
+
+Describe "Generate And Build SDK" -Tag "pipeline" {
+    BeforeAll {
+        # . $PSScriptRoot/../Invoke-GenerateAndBuildV2.ps1
+        # clean
+        $sdkRootPath =  (Join-Path $PSScriptRoot .. .. ..)
+        $sdkFolder = (Join-Path $sdkRootPath "sdk" "deviceupdate" "Azure.IoT.DeviceUpdate")
+        if (Test-Path $sdkFolder) {
+            Remove-Item $sdkFolder -Recurse
+        }
+        $artifactFolder = (Join-Path $sdkRootPath "artifacts" "packages" "Debug" "Azure.IoT.DeviceUpdate")
+        if (Test-Path $artifactFolder) {
+            Remove-Item $artifactFolder -Recurse
+        }
+    }
+
+    It "scenaro 1: first onboard" {
+        & $PSScriptRoot/../Invoke-GenerateAndBuildV2.ps1 -inputJsonFile $PSScriptRoot/firstOnboardInput.json -outputJsonFile $PSScriptRoot/firstOnboardOutput.json
+        $outputJson = Get-Content $PSScriptRoot/firstOnboardOutput.json | Out-String | ConvertFrom-Json
+
+        $outputJson.packages.Count | Should -Be 1
+        $outputJson.packages[0].packageName | Should -Be "Azure.IoT.DeviceUpdate"
+        $outputJson.packages[0].result | Should -Be "succeeded"
+        Remove-Item $PSScriptRoot/firstOnboardOutput.json
+    }
+
+    It "scenaro 2: update existing SDK without autorest config" {
+        & $PSScriptRoot/../Invoke-GenerateAndBuildV2.ps1 -inputJsonFile $PSScriptRoot/updateExistingInput.json -outputJsonFile $PSScriptRoot/updateExistingOutput.json
+        $outputJson = Get-Content $PSScriptRoot/updateExistingOutput.json | Out-String | ConvertFrom-Json
+
+        $outputJson.packages.Count | Should -Be 1
+        $outputJson.packages[0].packageName | Should -Be "Azure.IoT.DeviceUpdate"
+        $outputJson.packages[0].result | Should -Be "succeeded"
+    }
+
+    It "scenaro 2: update existing SDK with autorest config" {
+        & $PSScriptRoot/../Invoke-GenerateAndBuildV2.ps1 -inputJsonFile $PSScriptRoot/updateExistingWithConfigInput.json -outputJsonFile $PSScriptRoot/updateExistingWithConfigOutput.json
+        $outputJson = Get-Content $PSScriptRoot/updateExistingWithConfigOutput.json | Out-String | ConvertFrom-Json
```

Comment:
Since you have the output json already included in these changes, it may also be worth doing an object or text comparison of the expected output file and the one generated by this command.

Reply:
In the output json there are some values varying every time, such as the 'artifacts', the name artifacts will contain the generate-date and it will vary every time, So I cannot just compare the whole object or text.